src/app/static/src/app/components/adr/SelectDataset.vue

```diff
@@ -23,7 +23,7 @@
         <button
             class="btn btn-red"
             :class="selectedDataset && 'ml-2'"
-            @click="toggleModal"
+            @click="selectedDataset? confirmEditing($event) : toggleModal()"
```

Comment:
Curious about the inconsistency here... why do we pass `$event` to `confirmEditing()` but not `toggleModal()`? Why does one have `preventDefault()` but not the other?

More generally, this feels slightly convoluted. Would be keen to hear others' thoughts, but maybe `toggleModal()` should be replaced by `openModal()` and `closeModal()`, where the former includes a check for whether a dataset is already selected?